include/solvers/navier_stokes_scratch_data.h

```diff
@@ -797,36 +802,106 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
 
     fe_values_local_particles[pressure].get_function_gradients(
       previous_solution, fluid_pressure_gradients_at_particle_location);
+  }
 
-    // Create a quadrature for the void fraction that is based on the particle
-    // reference location
-    if (this->interpolated_void_fraction)
-      {
-        FEValues<dim> fe_values_particles_void_fraction(
-          this->fe_values_void_fraction->get_fe(), q_local, update_values);
+  /** @brief Interpolates the void fraction at the locations of the particles.
+   *
+   * @param q_particles Quadrature type object that contains the location of the particles
+   * relative to the cell's frame of reference.
+   *
+   * @param void_fraction_dof_handler The DoFHandler of the void fraction equations
+   *
+   * @param void_fraction_solution The void fraction calculated with one of the methods of
+   * the VoidFractionBase class.
+   */
+
+  template <typename VectorType>
+  void
+  calculate_void_fraction_at_particle_location(
+    const Quadrature<dim> &q_particles,
+    const DoFHandler<dim> &void_fraction_dof_handler,
+    const VectorType      &void_fraction_solution)
+  {
+    FEValues<dim> fe_values_particles_void_fraction(
+      this->fe_values_void_fraction->get_fe(), q_particles, update_values);
 
-        const auto &void_fraction_dh_cell =
-          typename DoFHandler<dim>::cell_iterator(
-            *this->fe_values_void_fraction->get_cell(),
-            &void_fraction_dof_handler);
+    const auto &void_fraction_dh_cell = typename DoFHandler<dim>::cell_iterator(
+      *this->fe_values_void_fraction->get_cell(), &void_fraction_dof_handler);
```

Comment:
I think this is ok for now, but this FEValues should really be a FEPointEvaluator instead of a regular FEValues. I would open an issue for this so that we keep track of this